File: JS Implementations/Queue Data Structure/Queue_Array.js
class Queue {
  constructor() {
    this.items = [];
  }

  // Add an element to the end of the queue
  enqueue(item) {
    this.items.push(item);
  }

  // Remove and return the element at the front of the queue
  dequeue() {
    if (this.isEmpty()) {
      return null;
    }
    return this.items.shift(); // shift removes the first element
  }

  // View the element at the front without removing it
  peek() {
    if (this.isEmpty()) {
      return null;
    }
    return this.items[0];
  }

  // Check if the queue is empty
  isEmpty() {
    return this.items.length === 0;
  }

  // Return the number of elements in the queue
  size() {
    return this.items.length;
  }
}

// Example usage:
const queue = new Queue();
queue.enqueue(10);
queue.enqueue(20);
console.log(queue.peek()); // 10
console.log(queue.dequeue()); // 10
console.log(queue.size()); // 1

File: Python Implementations/Queue Data Structure/Queue_Array.py
class Queue:
  def __init__(self):
    self.queue = []
    
  def enqueue(self, element):
    self.queue.append(element)

  def dequeue(self):
    if self.is_empty():
      return "Queue is empty"
    return self.queue.pop(0)

  def peek(self):
    if self.is_empty():
      return "Queue is empty"
    return self.queue[0]

  def is_empty(self):
    return len(self.queue) == 0

  def size(self):
    return len(self.queue)

# Create a queue
myQueue = Queue()

myQueue.enqueue('A')
myQueue.enqueue('B')
myQueue.enqueue('C')

print("Queue: ", myQueue.queue)
print("Peek: ", myQueue.peek())
print("Dequeue: ", myQueue.dequeue())
print("Queue after Dequeue: ", myQueue.queue)
print("is_empty: ", myQueue.is_empty())
print("Size: ", myQueue.size())